src/volume_mgr.rs

```diff
@@ -319,13 +353,55 @@ where
                 return Err(Error::VolumeStillInUse);
             }
         }
-
         let volume_idx = self.get_volume_by_id(volume)?;
+        if !self.open_volumes[volume_idx].read_only {
+            let VolumeType::Fat(volume_type) = &self.open_volumes[volume_idx].volume_type;
+            self.set_volume_status_dirty(volume_type, false)?;
+        }
+
         self.open_volumes.swap_remove(volume_idx);
 
         Ok(())
     }
 
+    /// Sets the volume status dirty to dirty if true, to not dirty if false
+    fn set_volume_status_dirty(
+        &self,
+        volume: &FatVolume,
```

Comment:
I think this should take `volume_idx` and handle the different types of volume internally. It should also skip updating the FAT if the FAT is already marked as dirty.

Reply:
fixed in 56a07fec25eb9af5f54b90b6e9d49f275962b149